language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/TypeGuardCodeAction.java

```diff
@@ -175,10 +172,31 @@ public String getName() {
         SemanticModel semanticModel = context.currentSemanticModel().orElseThrow();
         Document srcFile = context.currentDocument().orElseThrow();
         Optional<Symbol> symbol = semanticModel.symbol(srcFile,
-                                                       assignmentStmtNode.varRef().lineRange().startLine());
+                assignmentStmtNode.varRef().lineRange().startLine());
         if (symbol.isEmpty() || symbol.get().kind() != SymbolKind.VARIABLE) {
             return Optional.empty();
         }
         return Optional.of((VariableSymbol) symbol.get());
     }
+
+    private boolean isInValidContext(NonTerminalNode matchedNode, CodeActionContext context) {
+        Optional<SyntaxTree> syntaxTree = context.currentSyntaxTree();
+        if (syntaxTree.isEmpty()) {
+            return false;
+        }
+        Node node;
+        if (matchedNode.kind() == SyntaxKind.ASSIGNMENT_STATEMENT) {
+            node = ((AssignmentStatementNode) matchedNode).varRef();
+        } else if (matchedNode.kind() == SyntaxKind.LOCAL_VAR_DECL) {
+            node = ((VariableDeclarationNode) matchedNode).typedBindingPattern().bindingPattern();
+        } else {
+            return false;
+        }
+        return isCursorWithinIdentifier(node, context);
+    }
+
+    private boolean isCursorWithinIdentifier(Node node, CodeActionContext context) {
```

Comment:
Shall we move this to the `CommonUtil` as `isWithinRange` given the offset captured as the parameter